racket/src/thread/place-message.rkt

```diff
@@ -245,24 +261,36 @@
   (define direct-cyclic (read (open-input-string "#0=(1 #0# 2)")))
   (test #t (place-message-allowed-direct? direct-cyclic))
 
+  (test #t (place-message-allowed? (make-prefab-struct 'bx (box #f) (box #f))))
+  (test #f (place-message-allowed-direct? (make-prefab-struct 'bx (box #f) (box #f))))
+
   (define stateful-cyclic (make-reader-graph
                            (let ([ph (make-placeholder #f)]
                                  [ph2 (make-placeholder #f)]
                                  [ph3 (make-placeholder #f)])
                              (define (as ph v) (placeholder-set! ph v) v)
                              (as ph2 (vector (as ph (cons ph (string-copy "apple")))
+                                             (box (box 2))
                                              ph2
                                              (as ph3 (hasheq 'a 1 'b ph3))
                                              '#s(pre 4 5)))
                              ph)))
   (test #f (place-message-allowed-direct? stateful-cyclic))
   (test #t (place-message-allowed? stateful-cyclic))
-  (test stateful-cyclic (un-message-ize (message-ize stateful-cyclic)))
-
+  (test stateful-cyclic (un-message-ize (message-ize stateful-cyclic
+                                                     (lambda ()
```

Comment:
Why did you add the second argument here?

Reply:
I assumed `message-ize` referred to the definition on line 94 which appears to require a second argument. Have I missed something?

Reply:
Ah, the problem is that this test code is not run by default and thus was totally broken before your change. Sigh.

Reply:
That's actually encouraging, to me at least - I figured the same.